### PR TITLE
Point Verdant now only spawns when today is a Port of Call

### PR DESCRIPTION
**Pull request by Ben10083**
title

As it says above, unless it is a port of call day, PV will no longer spawn (it will always spawn and ignore budget if it is a Port day though)

New template flag was made to handle this and any future maps that would use this port of call system.

Haneunim Port of Call days changed to Saturday and Sunday

### CODE REVIEW

**Comment by Triogenix** [comment]
As I said in discord, this is currently being discussed internally by the lore team.

**Comment by Ben10083** [comment]
> As I said in discord, this is currently being discussed internally by the lore team.

Yep, making this PR so once the discussion is complete, this PR can move forward or adapt (if PV doesn't want to use this system, will likely just remove the flag from PV but leave the system in place)

**Comment by Triogenix** [comment]
We ended up deciding that we'll be moving the port of call days to Saturday-Sunday, and have PV able to spawn on those days as well. IDK how that fits with the PR you had, but if it could be updated to reflect that, it'll be good to be merged in terms of lore discussion stuff.

**Comment by Ben10083** [comment]
attempting to fix a bug that is making it not spawn when it should

**Comment by Ben10083** [comment]
I think it works now

**Comment by Triogenix** [comment]
👍 This is fine with the lore team then.

File: code/datums/ruins.dm [hunk 1]
```diff
@@ -86,3 +86,11 @@
 /// Returns `TRUE` if this ruin can spawn in current sector, otherwise `FALSE`.
 /datum/map_template/ruin/proc/spawns_in_current_sector()
 	return (SSatlas.current_sector.name in sectors)
+
+/datum/map_template/ruin/proc/port_of_call_check()
```
Comment (NonQueueingMatt) [inline code/datums/ruins.dm:90]: this shouldn't be a ruin proc, this should either be on SSatlas, on the mapdatum, or on the sector - probably on SSatlas

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/maps/ruins.dm [hunk 2]
```diff
@@ -36,6 +36,17 @@ GLOBAL_LIST_EMPTY(banned_ruin_ids)
 			handled_ruin_paths += ruin.type
 			continue
 
+		if((ruin.template_flags & TEMPLATE_FLAG_PORT_SPAWN) && (ruin.spawns_in_current_sector()))
```
Comment (FluffyGhoster) [inline code/modules/maps/ruins.dm:39]: The check below already handles the spawns_in_current_sector(), what is the point of processing it here too, over moving this block below it and just cycle the potential ruins that it already filtered?

**Review by NonQueueingMatt — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**